src/redux/investment/actionCreator.js

```diff
@@ -0,0 +1,89 @@
+import { SERVER_URL, API_URL, API_KEY } from 'react-native-dotenv';
+import {
+  FETCH_USER_INVESTMENT_SHARES_START,
+  FETCH_USER_INVESTMENT_SHARES_FAIL,
+  FETCH_USER_INVESTMENT_SHARES_SUCCESS,
+  FETCH_STOCK_PRICE_START,
+  FETCH_STOCK_PRICE_FAIL,
+  FETCH_STOCK_PRICE_SUCCESS,
+  FETCH_STOCK_PRICE,
+} from './actionType';
+import { round, transferToDollar } from '../../common/numbers';
+
+export const fetchUserInvestmentSharesStart = () => ({
+  type: FETCH_USER_INVESTMENT_SHARES_START,
+});
+
+export const fetchUserInvestmentSharesFail = (payload) => ({
+  type: FETCH_USER_INVESTMENT_SHARES_FAIL,
+  payload,
+});
+
+export const fetchUserInvestmentSharesSuccess = (payload) => ({
+  type: FETCH_USER_INVESTMENT_SHARES_SUCCESS,
+  payload,
+});
+
+export const fetchStockPriceStart = () => ({
+  type: FETCH_STOCK_PRICE_START,
+});
+
+export const fetchStockPriceFail = (payload) => ({
+  type: FETCH_STOCK_PRICE_FAIL,
+  payload,
+});
+
+export const fetchStockPriceSuccess = (payload) => ({
+  type: FETCH_STOCK_PRICE_SUCCESS,
+  payload,
+});
+
+export const fetchStockPrice = (stocks, symbols) => (dispatch) => {
+  const apiUrl = new URL(`${API_URL}/stock/market/batch`);
+  apiUrl.searchParams.append('symbols', symbols);
+  apiUrl.searchParams.append('types', 'price');
+  apiUrl.searchParams.append('token', API_KEY);
+  dispatch(fetchStockPriceStart());
+  fetch(apiUrl)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error('Oops, there\'s something wrong with our app.');
+      }
+      return response.json();
+    })
+    .then((response) => {
+      dispatch(fetchStockPriceSuccess(response));
+      const calculatedValue = stocks
+        .map(({ shares, symbol }) => shares * response[symbol].price)
+        .reduce((a, b) => a + b);
+      dispatch({
+        type: FETCH_STOCK_PRICE,
+        payload: transferToDollar(round(calculatedValue)),
+      });
```

Comment:
this part is a bit chaotic for me. Do you really need these two dispatches? I think, you don't really use the first one, or even if you use it, why don't you dispatch only one action with the total value and the given stock prices? Like:
```js
{
  totalValue: ...  // also you could calculate this in the reducer, but both are fine for me.
  stocks: [
    {
      symbol: 'FB',
      qty: 123,
      price: ...
    },
    ...
  ],
}
```
Also, you should use actionCreators always, do not dispatch a simple object.